build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,210 @@
+	/**
+	 * The type of release for the package (e.g., "alpha", "beta", "public", "default").
+	 */
+	releaseType: ReleaseTag;
+	/**
+	 * The JSON data for the package.
+	 */
+	json: PackageJson;
+}
+
+const knownReleaseTag = ["alpha", "beta", "public", "default"] as const;
+
+type ReleaseTag = typeof knownReleaseTag[number];
+
+function isReleaseTag(str: string | undefined): str is ReleaseTag {
+	return str === undefined ? false : knownReleaseTag.includes(str as any);
+}
+
+export default class SetReleaseTagPublishingCommand extends PackageCommand<
+	typeof SetReleaseTagPublishingCommand
+> {
+	static readonly description =
+		"Updates the .d.ts types that are included in the `types` field in package.json. This command is intended to be used in CI only.";
+
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		types: Flags.custom<ReleaseTag>({
+			description: "Which .d.ts types to include in the published package.",
+			required: true,
+			parse: async (input) => {
+				if (isReleaseTag(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private packageList!: PackageTypesList;
+
+	// Override the init method to initialize packageList
+	public async init() {
+		await super.init();
+		this.packageList = {
+			packagesNotUpdated: [],
+			packagesUpdated: [],
+		};
+	}
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		let packageUpdate: boolean = false;
+
+		updatePackageJsonFile(pkg.directory, (json) => {
+			const types: string | undefined = json.types ?? json.typings;
+
+			if (types === undefined) {
+				return;
+			}
+
+			const configOptions = ExtractorConfig.tryLoadForFolder({
+				startingFolder: pkg.directory,
+			});
+			if (configOptions === undefined) {
+				this.verbose(`No api-extractor config found for ${pkg.name}. Skipping.`);
+				return;
+			}
+			const extractorConfig = ExtractorConfig.prepare(configOptions);
+
+			const config: ReleaseTagConfig = {
+				pkg,
+				extractorConfig,
+				releaseType: this.flags.types as ReleaseTag,
+				json,
+			};
+
+			packageUpdate = updatePackageJsonTypes(config, this.logger);
+
+			updatePackageLists(this.packageList, packageUpdate, pkg);
+		});
+	}
+
+	public async run(): Promise<PackageTypesList> {
+		// Calls processPackage on all packages.
+		await super.run();
+
+		const flags = this.flags;
+
+		if (this.packageList.packagesUpdated.length === 0) {
+			this.log(`No updates in package.json for ${flags.types} release tag`);
+		}
+
+		return this.packageList;
+	}
+}
+
+/**
+ * Updates the types/typing field in package.json.
+ * @param config - An interface specifying the package, extractor config, release type, JSON data
+ * @param log - The logger used for logging verbose information.
+ * @returns true if the update was successful, false otherwise.
+ */
+function updatePackageJsonTypes(config: ReleaseTagConfig, log: CommandLogger): boolean {
```

Comment:
It's probably worth calling out that even though this function takes a Package and a PackageJson, the only changes that will be reflected in 

It might be clearer to take the package.json path (`pkg.packageJsonFileName`) as the argument instead of the whole package, since that's the only value you use.

For testability it would be nice if this changed the Package itself and wrote to the file separately. But that's probably a lot of refactoring that's not worth it.

Reply:
currently `pkg.directory` is passed instead of `pkg.packageJsonFileName`